src/test/java/io/cdap/plugin/sfmc/sink/MarketingCloudConfTest.java

```diff
@@ -0,0 +1,298 @@
+import io.cdap.cdap.etl.api.validation.CauseAttributes;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.sfmc.source.util.SourceObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import static io.cdap.plugin.sfmc.source.util.MarketingCloudConstants.PROPERTY_AUTH_API_ENDPOINT;
+import static io.cdap.plugin.sfmc.source.util.MarketingCloudConstants.PROPERTY_CLIENT_ID;
+import static io.cdap.plugin.sfmc.source.util.MarketingCloudConstants.PROPERTY_CLIENT_SECRET;
+import static io.cdap.plugin.sfmc.source.util.MarketingCloudConstants.PROPERTY_DATA_EXTENSION_KEY;
+import static io.cdap.plugin.sfmc.source.util.MarketingCloudConstants.PROPERTY_SOAP_API_ENDPOINT;
+
+public class MarketingCloudConfTest {
+
+  public static final String TEST_REF_NAME = "TestRefName";
+  protected static final String CLIENT_ID = "clientId";
+  protected static final String CLIENT_SECRET = "clientSecret";
+  protected static final String AUTH_ENDPOINT = "authEndPoint";
+  protected static final String SOAP_ENDPOINT = "soapEndPoint";
+  private static final String TEST_DATA_EXTENSION = "DE";
+  private static Schema schema;
+
+
+  @Test
+  public void testConfig() {
+    SourceObject object = SourceObject.DATA_EXTENSION;
+    MarketingCloudConf config = new MarketingCloudConf("referenceName", CLIENT_ID,
+      CLIENT_SECRET, "DE", AUTH_ENDPOINT,
+      SOAP_ENDPOINT, 500, null, "<key=value>",
+      true, true, true);
+    Assert.assertEquals("referenceName", config.getReferenceName());
+    Assert.assertEquals(500, config.getMaxBatchSize());
+    Assert.assertTrue(config.shouldFailOnError());
+    Assert.assertTrue(config.shouldReplaceWithSpaces());
+    Assert.assertTrue(config.shouldTruncateText());
+    Assert.assertEquals(Operation.INSERT, config.getOperation());
+    Assert.assertEquals("Data Extension", object.getValue());
+    Assert.assertEquals("Data Extension, Email, Mailing List, Bounce Event, Open Event, Unsub Event, Sent Event, " +
+      "Notsent Event", SourceObject.getSupportedObjects());
+  }
+
+  @Test
+  public void testValidateReferenceName() {
+    MarketingCloudConf config = MarketingCloudConfHelper.newConfigBuilder().setReferenceName(TEST_REF_NAME)
+      .build();
+    Assert.assertEquals("TestRefName", config.getReferenceName());
+
+  }
+
+  @Test
+  public void testValidateClientIdNull() {
+    MockFailureCollector collector = new MockFailureCollector();
+    MarketingCloudConf config = withSalesforceValidationMock(MarketingCloudConfHelper.newConfigBuilder()
+
+      .setClientId(null)
+      .setClientSecret(CLIENT_SECRET)
+      .setAuthEndpoint(AUTH_ENDPOINT)
+      .setSoapEndpoint(SOAP_ENDPOINT)
+      .build(), collector);
+
+    try {
+      config.validate(schema, collector);
+      collector.getOrThrowException();
```

Comment:
fail test if exception is not thrown. same comment for other test cases

Reply:
Done